src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -555,6 +618,27 @@ void ProducerEventProcessorImpl::ProcessWarningInstrumentingWithUprobesEventAndT
   client_capture_event_collector_->AddEvent(std::move(event));
 }
 
+void ProducerEventProcessorImpl::ProcessThreadStateSliceCallstack(
+    ThreadStateSliceCallstack* thread_state_slice_callstack) {
+  const Callstack& callstack = thread_state_slice_callstack->callstack();
+
+  std::pair<std::vector<uint64_t>, Callstack::CallstackType> callstack_data{
+      {callstack.pcs().begin(), callstack.pcs().end()}, callstack.type()};
+  auto [callstack_id, assigned] = callstack_pool_.GetOrAssignId(callstack_data);
+
+  if (assigned) {
+    ClientCaptureEvent interned_callstack_event;
+    interned_callstack_event.mutable_interned_callstack()->set_key(callstack_id);
+    interned_callstack_event.mutable_interned_callstack()->set_allocated_intern(
+        thread_state_slice_callstack->release_callstack());
+    client_capture_event_collector_->AddEvent(std::move(interned_callstack_event));
+  }
+
+  thread_state_slice_tid_and_begin_timestamp_to_callstack_id_[{
```

Comment:
I think we should add a comment explaining how the interned callstacks are sent right away and only the id is kept for the matching with the slice.

Reply:
Done.